lib/query/country_info.rb

```diff
@@ -39,7 +42,7 @@ def results
     end
 
     def legislatures
-      results.map { |i| i[:legislature] }.uniq
+      results.delete_if { |result| result[:legislatureEnd] }.map { |i| i[:legislature] }.uniq
```

Comment:
methods that mutate global state are generally very suspect, as now other methods can return different results depending on the order of calling things. If you need to filter something, it's usually much better to have a method that does that filtering but leaves the underlying state intact.

Reply:
Eek, I didn't consider that it would be mutating global state :grimacing:

Reply:
AIUI `delete_if` is essentially equivalent to `reject!`, and I always peer very closely at any `!` methods.